File: src/countGoodNodesInBinaryTree..py
class TreeNode:
     def __init__(self, val=0, left=None, right=None):
         self.val = val
         self.left = left
         self.right = right


class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        goodNodesCount = 0

        def helper(root, prevVal):
            nonlocal goodNodesCount
            if root:
                if root.val >= prevVal:
                    goodNodesCount += 1

                helper(root.left, max(root.val, prevVal))
                helper(root.right, max(root.val, prevVal))

        helper(root, root.val)

        return goodNodesCount


if __name__ == '__main__':
    root = TreeNode(3)
    root.left = TreeNode(1)
    root.left.left = TreeNode(3)
    root.right = TreeNode(4)
    root.right.left = TreeNode(1)
    root.right.right = TreeNode(5)

    sol = Solution()
    print(sol.goodNodes(root))

File: src/countGoodNodesInBinaryTree.java

class Solution {
    private int goodNodes = 0;
    public int helper(TreeNode root, int max) {
        if(root != null){
            if(root.val >= max){
                goodNodes++;
                max = root.val;
            }
            helper(root.left, max);
            helper(root.right, max);
        }
        return goodNodes;
    }
    public int goodNodes(TreeNode root) {
        int result = helper(root, root.val);
        return result;
    }
}
